main/php_syslog.c

```diff
@@ -56,6 +56,16 @@ PHPAPI void php_syslog(int priority, const char *format, ...) /* {{{ */
 	smart_string sbuf = {0};
 	va_list args;
 
+	/*
+	 * don't rely on openlog() being called by syslog() if it's
+	 * not already been done; call it ourselves and pass the
+	 * correct parameters!
+	 */
+	if (!PG(have_called_openlog)) {
```

Comment:
What's the actually point of preventing repeated call to openlog? This should be always handled by the libc where multiple calls are ignored. I just checked [glibc syslog](https://sourceware.org/git/?p=glibc.git;a=blob;f=misc/syslog.c;h=644dbe80ec0b90d208515c95a53ba2d1ea86dd74;hb=HEAD#l343) and [musl syslog](https://github.com/esmil/musl/blob/194f9cf93da8ae62491b7386edf481ea8565ae4e/src/misc/syslog.c#L69) and both handle it as expected. I really don't think we need to introduce a global for such a minimal overhead.

Reply:
> What's the actually point of preventing repeated call to openlog?

That's not what it's doing.

It's calling openlog() if it hasn't already been called, with the default arguments.

If it has been called already (particularly with *non-*default arguments), then this stops it from clobbering the previous explicit invocation.

For instance, by default, it calls `openlog()` with `option` set to 0.  What if I had called:

`openlog(..., LOG_CONS | LOG_PID, ...);`

do I really want the next call to `syslog()` resetting the value of `option`?

Reply:
Ok I see. It makes sense I guess